actions/submit-signing-request/task.ts

```diff
@@ -231,32 +232,35 @@ export class Task {
             }
         });
 
-        const targetFilePath = path.join(process.env.GITHUB_WORKSPACE as string, this.signedArtifactDestinationPath)
+        const targetDirectory = this.resolveOrCreateDirectory(this.outputArtifactDirectory);
 
-        // make sure that the target directory exists
-        const targetDir = path.dirname(targetFilePath);
-        if (!fs.existsSync(targetDir)) {
-            core.info(`Directory ${targetDir} does not exist, and will be created`);
-            fs.mkdirSync(targetDir, { recursive: true });
-        }
+        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetDirectory}`);
 
-        core.info(`The signed artifact is being downloaded from SignPath and will be saved to ${targetFilePath}`);
-        const writer = fs.createWriteStream(targetFilePath)
+        // save the signed artifact to temp ZIP file
+        const tmpZipFile = path.join(targetDirectory, '__signpath_signed_artifact_tmp.zip');
+        const writer = fs.createWriteStream(tmpZipFile)
         response.data.pipe(writer);
         await new Promise((resolve, reject) => {
-            writer.on('finish', resolve)
-            writer.on('error', reject)
+             writer.on('finish', resolve)
+             writer.on('error', reject)
         });
 
-        core.info("The signed artifact has been successfully downloaded from SignPath");
-        return targetFilePath;
-    }
+        // unzip temp ZIP file to the targetDirectory
+        const zip = new nodeStreamZip.async({ file: tmpZipFile });
+        await zip.extract(null, targetDirectory);
 
-    async logArtifactFileStat(artifactPath: string) {
-        await fs.stat(artifactPath, (err, stats) => {
-            const size = fileSize.partial({base: 2, standard: "jedec"});
-            core.info("File size: " + size(stats.size));
-        });
+        // delete temp ZIP file
+        fs.unlinkSync(tmpZipFile);
```

Comment:
Should be cleaned up even if an error occurs above, imo

Reply:
added finally block

Reply:
eventually removed this block of code. Now rely on the agent and let it remove our temp files